src/coreclr/vm/syncblk.h

```diff
@@ -729,50 +729,29 @@ class InteropSyncBlockInfo
 #endif // FEATURE_COMINTEROP
 
 #if !defined(DACCESS_COMPILE)
-    // set m_pUMEntryThunkOrInterceptStub if not already set - return true if not already set
+    // set m_pUMEntryThunk if not already set - return true if not already set
     bool SetUMEntryThunk(void* pUMEntryThunk)
     {
         WRAPPER_NO_CONTRACT;
-        return (FastInterlockCompareExchangePointer(&m_pUMEntryThunkOrInterceptStub,
+        return (FastInterlockCompareExchangePointer(&m_pUMEntryThunk,
                                                     pUMEntryThunk,
                                                     NULL) == NULL);
     }
 
-    // set m_pUMEntryThunkOrInterceptStub if not already set - return true if not already set
-    bool SetInterceptStub(Stub* pInterceptStub)
-    {
-        WRAPPER_NO_CONTRACT;
-        void *pPtr = (void *)((UINT_PTR)pInterceptStub | 1);
-        return (FastInterlockCompareExchangePointer(&m_pUMEntryThunkOrInterceptStub,
-                                                    pPtr,
-                                                    NULL) == NULL);
-    }
-
-    void FreeUMEntryThunkOrInterceptStub();
+    void FreeUMEntryThunk();
 
 #endif // DACCESS_COMPILE
 
     void* GetUMEntryThunk()
     {
         LIMITED_METHOD_CONTRACT;
-        return (((UINT_PTR)m_pUMEntryThunkOrInterceptStub & 1) ? NULL : m_pUMEntryThunkOrInterceptStub);
-    }
-
-    Stub* GetInterceptStub()
-    {
-        LIMITED_METHOD_CONTRACT;
-        return (((UINT_PTR)m_pUMEntryThunkOrInterceptStub & 1) ? (Stub *)((UINT_PTR)m_pUMEntryThunkOrInterceptStub & ~1) : NULL);
+        return (((UINT_PTR)m_pUMEntryThunk & 1) ? NULL : m_pUMEntryThunk);
```

Comment:
Do we still set the low order bit during delete?

Reply:
The `SetInterceptStub` API did above, but about to the thunk itself?

Reply:
We don't. I'll clean this up.